volume/drivers/coprhd/coprhd.go

```diff
@@ -186,18 +272,123 @@ func (d *driver) Alerts(volumeID string) (*api.Alerts, error) {
 }
 
 func (d *driver) Attach(volumeID string) (path string, err error) {
-	return "", nil
+	v, err := d.GetVol(volumeID)
+	if err != nil {
+		return "", fmt.Errorf("Volume %s could not be located", volumeID)
+	}
+
+	vol, err := d.coprhd.Volume().
+		WWN(volumeID).
+		Query()
+	if err != nil {
+		return "", err
+	}
+
+	group, err := d.coprhd.Export().
+		Name(vol.Name).
+		Initiators(d.itr.Id).
+		Volumes(vol.Id).
+		Project(vol.Project.Id).
+		Array(vol.VArray.Id).
+		Create()
+	if err != nil {
+		return "", err
+	}
+
+	proto := d.itr.Protocol
+	dev := ""
+
+	switch proto {
+	case coprhd.InitiatorTypeScaleIO:
+		dev, err = d.waitAttachDevice(volumeID, time.Second*180)
+		if err != nil {
+			return "", err
+		}
+	default:
+		return "", fmt.Errorf("Unsupported initiator protocol")
+	}
+
+	v.AttachedOn = group.Id
+	v.DevicePath = dev
+	v.Status = api.VolumeStatus_VOLUME_STATUS_UP
+
+	err = d.UpdateVol(v)
+	if err != nil {
+		return "", err
+	}
+	return dev, nil
 }
 
 func (d *driver) Detach(volumeID string) error {
+	v, err := d.GetVol(volumeID)
```

Comment:
https://github.com/libopenstorage/openstorage/blob/master/STYLEGUIDE.md

"Variable names should reflect the type, ie an instance of Runner should be runner. This is in contrast to golang's official recommendation, but has been found to make code more readable. Heh. So this means api.Volume is not v, vol, whatever, it's volume, a request is not req, createReq, r, it's request. Only exception is the receiver argument on a function, ie func (s *server) Foo(...) { ... }."